File: src/main/java/com/GoldenMine/object/ObjectStorage.java
package com.GoldenMine.object;

import java.util.HashMap;

public class ObjectStorage {
    public static final ObjectStorage INSTANCE = new ObjectStorage();

    private HashMap<String, KoreanObjectCreator> creators = new HashMap<>();

    private ObjectStorage() {
        init();
    }

    private void init() {
        addCreator(new Point2DCreator());
    }

    public void addCreator(KoreanObjectCreator creator) {
        creators.put(creator.getType(), creator);
    }
}

File: src/main/java/kr/goldenmine/object/ObjectStorage.kt
package kr.goldenmine.`object`

import java.util.*

class ObjectStorage private constructor() {

    val creators = HashMap<String, KoreanObjectCreator>()

    init {
        init()
    }

    private fun init() {
        addCreator(Point2DCreator())
    }

    fun addCreator(creator: KoreanObjectCreator) {
        creators[creator.type] = creator
    }

    fun getCreator(str: String) : KoreanObjectCreator? {
        return creators[str]
    }

    companion object {
        val INSTANCE = ObjectStorage()
    }
}
